src/document/doc-builders.js

```diff
@@ -140,8 +140,15 @@ function conditionalGroup(states, opts) {
  */
 function fill(parts) {
   if (process.env.NODE_ENV !== "production") {
-    for (const part of parts) {
+    for (const [index, part] of parts.entries()) {
       assertDoc(part);
+      if (index % 2 === 1 && part.type !== "line") {
```

Comment:
It's not that simple. It also can be something like `["{' '}", softlline]`. The idea is that it contains a line break.

Reply:
Did you mean `["{", "}", softline]`?

Reply:
Anything with a line break inside

Reply:
Oh, you mean "elements with odd indices" only need a line break. 

```js
{
	  type: "fill", 
	  parts: [
		    "a", 
		    ["{' '}", softlline], // <- this
		    "b"
	  ]
}
```

Got it.

Reply:
Yes. Even-indexed ones can contain anything, but odd-indexed must contain at least one line break.